mesonbuild/options.py

```diff
@@ -233,12 +246,15 @@ def without_module_prefix(self) -> 'OptionKey':
             return self.evolve(newname)
         return self
 
+    def is_for_build(self) -> bool:
+        return self.machine == MachineChoice.BUILD
```

Comment:
you should always use identity comparisons with enums, otherwise you end up with an extra layer of indirection.